lib/astarte_data_access/kv_store.ex

```diff
@@ -89,6 +89,10 @@ defmodule Astarte.DataAccess.KvStore do
     query = from(__MODULE__, select: ^value_expr)
     primary_key = [group: group, key: key]
 
-    Repo.fetch_by(query, primary_key, opts)
+    case Repo.fetch_by(query, primary_key, opts) do
+      nil -> {:error, nil}
+      row -> {:ok, row}
+      {:error, error} -> {:error, error}
```

Comment:
the CI is already suggesting this but what I think you want to do here is
```suggestion
      {:error, error} -> {:error, error}
      row -> {:ok, row}
```
as otherwise the first row always matches